src/Grpc.Net.Client/Internal/GrpcProtocolHelpers.cs

```diff
@@ -91,6 +83,32 @@ public static Metadata BuildMetadata(HttpResponseHeaders responseHeaders)
             return headers;
         }
 
+        private static bool ShouldSkipHeader(string name)
+        {
+            if (name.Length == 0)
+            {
+                return false;
+            }
+
+            switch (name[0])
+            {
+                case ':':
+                    // ASP.NET Core includes pseudo headers in the set of request headers
+                    // whereas, they are not in gRPC implementations. We will filter them
+                    // out when we construct the list of headers on the context.
+                    return true;
+                case 'g':
+                case 'G':
+                    // Exclude known grpc headers. This matches Grpc.Core client behavior.
+                    return string.Equals(name, GrpcProtocolConstants.StatusTrailer, StringComparison.OrdinalIgnoreCase)
```

Comment:
What's the benefit of excluding these? I get that this matches, but why?

Reply:
Status/message have a strongly typed API for accessing them, and encoding/accept-encoding are automatically used for you by the client.

Reply:
Cool beans, but why is it **better** that they don't appear in the header collection? It surprises the heck out of me 😆

Reply:
Does it work this way for non-metadata responses?

Reply:
> Cool beans, but why is it **better** that they don't appear in the header collection? It surprises the heck out of me 😆

My guess is they didn't want to confuse users by showing metadata that the user didn't add on the server. I don't have a strong opinion, but since Grpc.Core does this, and compatibility with Grpc.Core is a goal, we'll follow their lead until there is a reason not to.

Reply:
Understood. IMO this would make perfect sense if this was called metadata instead of headers.